### PR TITLE
Build grpc-haskell on Windows

### PR DESCRIPTION
**Pull request by aherrmann-da**
- Patches `unix-compat` to work around missing version macros in hsc, see https://github.com/tweag/rules_haskell/issues/821.
- Inlines the `c2hs` generated files for `grpc-haskell-core`, as `c2hs` fails on Windows with
    ```
    c2hs.exe: Error during preprocessing custom header file
    ```
- Adds `grpc-haskell` to Windows CI.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/docs/source/support/release-notes.rst), if appropriate

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: 3rdparty/haskell/unix-compat.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
+--- src/System/PosixCompat/Files.hsc    2018-08-23 02:44:16.000000000 +0200
```
Comment (neil-da) [inline 3rdparty/haskell/unix-compat.patch]: This seems an odd place to put the patch. Should we have a `patches` directory somewhere? I'd much rather all our patches were in one place.

File: nix/third-party/gRPC-haskell/core/BUILD.bazel [hunk 2]
```diff
@@ -11,24 +11,12 @@ load(
 )
 load("//bazel_tools:fat_cc_library.bzl", "fat_cc_library")
 
-c2hs_suite(
+da_haskell_library(
```
Comment (neil-da) [inline nix/third-party/gRPC-haskell/core/BUILD.bazel]: Can we have a comment and fixme that we generated the .hs files manually here. (I can't think of a better place to put it)

File: nix/third-party/gRPC-haskell/core/chs/Network/GRPC/Unsafe.chi [hunk 3]
```diff
@@ -0,0 +1,2 @@
+C->Haskell Interface Version C->Haskell Compiler, version 0.28.6 Switcheroo, 25 November 2017
```
Comment (neil-da) [inline nix/third-party/gRPC-haskell/core/chs/Network/GRPC/Unsafe.chi]: Why are we checking in the .chi files? Aren't then unnecessary is we check in the .hs?

File: 3rdparty/haskell/unix-compat.patch (hunk 1 above)
Comment (aherrmann-da) [reply 3rdparty/haskell/unix-compat.patch]: Good point, I've moved it to where all the other Bazel related patches are.

File: nix/third-party/gRPC-haskell/core/BUILD.bazel (hunk 2 above)
Comment (aherrmann-da) [reply nix/third-party/gRPC-haskell/core/BUILD.bazel]: Yes, I've added a comment.

File: nix/third-party/gRPC-haskell/core/chs/Network/GRPC/Unsafe.chi (hunk 3 above)
Comment (aherrmann-da) [reply nix/third-party/gRPC-haskell/core/chs/Network/GRPC/Unsafe.chi]: I removed them.

**Pull request merged**